### PR TITLE
ci: Add Cloudwatch Policy on Karpetner nodes for Windows nodes

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Add Cloudwatch policy to scrape container logs from Karpenter launched nodes. This mainly to pull Kubelet logs from Windows nodes to track down flaky failures on Node joining failures. 
- Link on imported kubelet logs:https://aws.amazon.com/blogs/containers/fluent-bit-integration-in-cloudwatch-container-insights-for-eks/

**How was this change tested?**
- `/karpenter snapshot`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

File: .github/actions/e2e/setup-cluster/action.yaml [hunk 1]
```diff
@@ -115,7 +116,8 @@ runs:
             roleName: karpenter-irsa-${{ inputs.cluster_name }}
             permissionsBoundaryARN: "arn:aws:iam::${{ inputs.account_id }}:policy/GithubActionsPermissionsBoundary"
             permissionPolicyARNs:
-              - "arn:aws:iam::${{ inputs.account_id }}:policy/KarpenterControllerPolicy-${{ inputs.cluster_name }}""""
+              - "arn:aws:iam::${{ inputs.account_id }}:policy/KarpenterControllerPolicy-${{ inputs.cluster_name }}"
+              - "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy""""
```
Comment (jonathan-innis) [inline .github/actions/e2e/setup-cluster/action.yaml:120]: This wasn't succeeding up to this point?

Comment (engedaam) [reply .github/actions/e2e/setup-cluster/action.yaml:120]: It was succeeding to pull Karpenter logs and container logs on MNG nodes. This is to pull kubelet logs from the Karpenter nodes. I was wanting to pull kubelet logs for the windows failures.

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**